File: src/Application Specific/Classes/Entity Collections/ContractsOriginal/ContractOriginalSiteAsset/ContractOriginalSiteAssetSQL.cs
using System.Data;

namespace FSM.Entity
{
    namespace ContractOriginalSiteAssets
    {
        public class ContractOriginalSiteAssetSQL
        {
            private Sys.Database _database;

            public ContractOriginalSiteAssetSQL(Sys.Database database)
            {
                _database = database;
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public DataView GetAll_ContractSiteID(int ContractSiteID, int SiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@ContractSiteID", ContractSiteID, true);
                _database.AddParameter("@SiteID", SiteID, true);
                var dt = _database.ExecuteSP_DataTable("ContractOriginalSiteAsset_GetAll_ContractSiteID");
                dt.TableName = Sys.Enums.TableNames.tblContractSiteAsset.ToString();
                return new DataView(dt);
            }

            public ContractOriginalSiteAsset Insert(ContractOriginalSiteAsset oContractSiteAsset)
            {
                _database.ClearParameter();
                AddContractSiteAssetParametersToCommand(ref oContractSiteAsset);
                oContractSiteAsset.SetContractSiteAssetID = Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("ContractOriginalSiteAsset_Insert"));
                oContractSiteAsset.Exists = true;
                return oContractSiteAsset;
            }

            public void Delete(int ContractSiteID)
            {
                _database.ClearParameter();
                _database.AddParameter("@ContractSiteID", ContractSiteID, true);
                _database.ExecuteSP_NO_Return("ContractOriginalSiteAsset_Delete");
            }

            private void AddContractSiteAssetParametersToCommand(ref ContractOriginalSiteAsset oContractSiteAsset)
            {
                {
                    var withBlock = _database;
                    withBlock.AddParameter("@ContractSiteID", oContractSiteAsset.ContractSiteID, true);
                    withBlock.AddParameter("@AssetID", oContractSiteAsset.AssetID, true);
                    withBlock.AddParameter("@PricePerVisit", oContractSiteAsset.PricePerVisit, true);
                    withBlock.AddParameter("@PrimaryAsset", oContractSiteAsset.PrimaryAsset, true);
                    withBlock.AddParameter("@Type", oContractSiteAsset.Type, true);
                    withBlock.AddParameter("@Product", oContractSiteAsset.Product, true);
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}

File: src/Application Specific/Classes/Entity Collections/ContractsOriginal/ContractOriginalSiteAsset/ContractOriginalSiteAssetSQL.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace ContractOriginalSiteAssets

        Public Class ContractOriginalSiteAssetSQL
            Private _database As Sys.Database

            Public Sub New(ByVal database As Sys.Database)
                _database = database
            End Sub

#Region "Functions"

            Public Function GetAll_ContractSiteID(ByVal ContractSiteID As Integer, ByVal SiteID As Integer) As DataView
                _database.ClearParameter()
                _database.AddParameter("@ContractSiteID", ContractSiteID, True)
                _database.AddParameter("@SiteID", SiteID, True)
                Dim dt As DataTable = _database.ExecuteSP_DataTable("ContractOriginalSiteAsset_GetAll_ContractSiteID")
                dt.TableName = Entity.Sys.Enums.TableNames.tblContractSiteAsset.ToString
                Return New DataView(dt)
            End Function

            Public Function Insert(ByVal oContractSiteAsset As ContractOriginalSiteAsset) As ContractOriginalSiteAsset
                _database.ClearParameter()
                AddContractSiteAssetParametersToCommand(oContractSiteAsset)

                oContractSiteAsset.SetContractSiteAssetID = Entity.Sys.Helper.MakeIntegerValid(_database.ExecuteSP_OBJECT("ContractOriginalSiteAsset_Insert"))
                oContractSiteAsset.Exists = True
                Return oContractSiteAsset
            End Function

            Public Sub Delete(ByVal ContractSiteID As Integer)
                _database.ClearParameter()
                _database.AddParameter("@ContractSiteID", ContractSiteID, True)
                _database.ExecuteSP_NO_Return("ContractOriginalSiteAsset_Delete")
            End Sub

            Private Sub AddContractSiteAssetParametersToCommand(ByRef oContractSiteAsset As ContractOriginalSiteAsset)
                With _database
                    .AddParameter("@ContractSiteID", oContractSiteAsset.ContractSiteID, True)
                    .AddParameter("@AssetID", oContractSiteAsset.AssetID, True)
                    .AddParameter("@PricePerVisit", oContractSiteAsset.PricePerVisit, True)
                    .AddParameter("@PrimaryAsset", oContractSiteAsset.PrimaryAsset, True)
                    .AddParameter("@Type", oContractSiteAsset.Type, True)
                    .AddParameter("@Product", oContractSiteAsset.Product, True)

                End With
            End Sub

#End Region

        End Class

    End Namespace

End Namespace


